### ISSUE TITLE
When windows server ltsc2022 container is newly configured with WinSystemLocale as ja-JP, and the container exits, the windows server host shuts down.

### ISSUE THREAD

**Issue by gadgetwhiz** [microsoft/Windows-Containers#343]
**Observed**: When windows server ltsc2022 container is newly configured with WinSystemLocale as ja-JP, and the container exits, the windows server host shuts down.

**Expected**: Docker container termination should have no effect on docker parent host.

Base state: Windows Server 2022 VM with Docker Desktop installed

**Steps to reproduce:**
- Access Windows Server 2022 host via Remote Desktop as admin
- Pull image, launch container (CLI): 
```
docker pull mcr.microsoft.com/windows/servercore:ltsc2022
docker image ls
#mcr.microsoft.com/windows/servercore   ltsc2022   d84fae21da7a   2 weeks ago   3.92GB
docker run -it mcr.microsoft.com/windows/servercore:ltsc2022 powershell
```
- Append WinUserLanguageList, set WinSystemLocale
```
Get-WinUserLanguageList
# OBSERVE: en-US in list

$UserLangList = Get-WinUserLanguageList
$UserLangList.Add("ja-JP")
$UserLangList
# OBSERVE: en-US and ja-JP in list

Set-WinUserLanguageList -LanguageList $UserLangList -Force 
Get-WinUserLanguageList
# OBSERVE: en-US and ja in list (NOT ja-JP)

Get-WinSystemLocale
Set-WinSystemLocale -SystemLocale ja-JP
Get-WinSystemLocale
# OBSERVE: en-US shown (NOT ja-JP) (** NOTE: this is non-deterministic as the language may occasionally change to ja-JP WITHOUT exit/start)
exit

OBSERVE: clean container exit if winsystemlocale is en-US, intermittent server shutdown if winsystemlocale is ja-JP
```
- Get container ID, restart, and connect:
```
docker container ls -a
docker start CONTAINER_ID
docker attach CONTAINER_ID

Get-WinSystemLocale
Set-WinSystemLocale -SystemLocale ja-JP
Get-WinSystemLocale
OBSERVE: ja-JP  (** NOTE: this is non-deterministic as the language does not always change)
exit

OBSERVE: long pause, Reconnecting 'The connection has been lost'. This may not happen every time.
```
- Relaunch remote desktop:
```
#OBSERVE: Shutdown Event Tracker dialog and prompt
```
- Event Viewer Summary
```
Critical, Event ID 41, Kernel-Power:
The system has rebooted without cleanly shutting down first. This error could be caused if the system stopped responding, crashed, or lost power unexpectedly.
```
- Event Viewer Details:
```
+ System 

  - Provider 

   [ Name]  Microsoft-Windows-Kernel-Power 
   [ Guid]  {331c3b3a-2005-44c2-ac5e-77220c37d6b4} 
 
   EventID 41 
 
   Version 8 
 
   Level 1 
 
   Task 63 
 
   Opcode 0 
 
   Keywords 0x8000400000000002 
 
  - TimeCreated 

   [ SystemTime]  2023-03-27T13:45:23.2544398Z 
 
   EventRecordID 5558 
 
   Correlation 
 
  - Execution 

   [ ProcessID]  4 
   [ ThreadID]  8 
 
   Channel System 
 
   Computer *****
 
  - Security 

   [ UserID]  S-1-5-18 
 

- EventData 

  BugcheckCode 215 
  BugcheckParameter1 0xfffff80367b00000 
  BugcheckParameter2 0x1 
  BugcheckParameter3 0x0 
  BugcheckParameter4 0x0 
  SleepInProgress 0 
  PowerButtonTimestamp 0 
  BootAppStatus 0 
  Checkpoint 0 
  ConnectedStandbyInProgress true 
  SystemSleepTransitionsToOn 0 
  CsEntryScenarioInstanceId 4 
  BugcheckInfoFromEFI false 
  CheckpointStatus 0 
  CsEntryScenarioInstanceIdV2 4 
  LongPowerButtonPressDetected false 
```
- systeminfo
```
C:\Users\Administrator>systeminfo

Host Name:                 *****
OS Name:                   Microsoft Windows Server 2022 Standard
OS Version:                10.0.20348 N/A Build 20348
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Server
OS Build Type:             Multiprocessor Free
Registered Owner:          Windows User
Registered Organization:
Product ID:                00454-20165-46881-AA047
Original Install Date:     1/30/2023, 11:04:21 AM
System Boot Time:          3/26/2023, 10:41:54 AM
System Manufacturer:       VMware, Inc.
System Model:              VMware7,1
System Type:               x64-based PC
Processor(s):              2 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 85 Stepping 4 GenuineIntel ~1995 Mhz
                           [02]: Intel64 Family 6 Model 85 Stepping 4 GenuineIntel ~1995 Mhz
BIOS Version:              VMware, Inc. VMW71.00V.16707776.B64.2008070230, 8/7/2020
Windows Directory:         C:\Windows
System Directory:          C:\Windows\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-06:00) Central Time (US & Canada)
Total Physical Memory:     16,383 MB
Available Physical Memory: 13,176 MB
Virtual Memory: Max Size:  18,815 MB
Virtual Memory: Available: 15,781 MB
Virtual Memory: In Use:    3,034 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\*****
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB5022507
                           [02]: KB5012170
                           [03]: KB5023705
                           [04]: KB5023793
Network Card(s):           1 NIC(s) Installed.
                           [01]: Intel(R) *****
                                 Connection Name: Ethernet0
                                 DHCP Enabled:    Yes
                                 DHCP Server:     *****
                                 IP address(es)
                                 [01]: *****
                                 [02]: *****
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```
- docker info
```
C:\Users\Administrator>docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.10.3)
  compose: Docker Compose (Docker Inc., v2.15.1)
  dev: Docker Dev Environments (Docker Inc., v0.1.0)
  extension: Manages Docker extensions (Docker Inc., v0.2.18)
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc., 0.6.0)
  scan: Docker Scan (Docker Inc., v0.25.0)
  scout: Command line tool for Docker Scout (Docker Inc., v0.6.0)

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 2
 Server Version: 20.10.23
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: process
 Kernel Version: 10.0 20348 (20348.1.amd64fre.fe_release.210507-1500)
 Operating System: Windows Server 2022 Standard Version 2009 (OS Build 20348.1607)
 OSType: windows
 Architecture: x86_64
 CPUs: 4
 Total Memory: 16GiB
 Name: *****
 ID: *****
 Docker Root Dir: C:\ProgramData\Docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Comment by gadgetwhiz** [comment]
UPDATE: I also sent a note to Docker Support so they were aware of the issue and any role that Docker may play in the issue. I received a reply that pointed out that Docker Desktop on Windows Server is not eligible for support and recommended I use docker engine instead. 

I reset my VM to a clean snapshot and installed Docker Engine according to their links/instructions. I am able to duplicate the server shutdown with docker engine.

Updated docker info:
```
C:\Users\Administrator>docker info
Client:
 Context:    default
 Debug Mode: false

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 1
 Server Version: 23.0.1
 Storage Driver: windowsfilter
  Windows:
 Logging Driver: json-file
 Plugins:
  Volume: local
  Network: ics internal l2bridge l2tunnel nat null overlay private transparent
  Log: awslogs etwlogs fluentd gcplogs gelf json-file local logentries splunk syslog
 Swarm: inactive
 Default Isolation: process
 Kernel Version: 10.0 20348 (20348.1.amd64fre.fe_release.210507-1500)
 Operating System: Microsoft Windows Server Version 21H2 (OS Build 20348.1607)
 OSType: windows
 Architecture: x86_64
 CPUs: 4
 Total Memory: 16GiB
 Name: **********
 ID: 2827669a-9355-4a72-8e98-df9421048ddd
 Docker Root Dir: C:\ProgramData\docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
 Product License: Community Engine
```

**Comment by fady-azmy-msft** [comment]
Thank you for highlighting this issue @gadgetwhiz. We were able to repro this issue and identify the root cause, and are working to publish the fix. I don't have a timeline on this yet but will report back when I do.

**Comment by a09-capf** [comment]
Hi @fady-azmy-msft,

I face the same issue.
Is there any update?

**Comment by fady-azmy-msft** [comment]
I'm afraid I don't have any updates to share, we're still looking into this.

**Comment by a09-capf** [comment]
Thank you for letting me know. I look forward to the issue being resolved.

**Comment by fady-azmy-msft** [comment]
I can't share timelines but the fix is on track to coming out soon.

**Comment by NAWhitehead** [comment]
@a09-capf, this issue has been resolved and was released in the 2023.10 B release.
